Subject: x86: kprobes: Remove remove bogus preempt_enable
From: Thomas Gleixner <tglx@linutronix.de>
Date: Thu, 17 Mar 2011 11:02:15 +0100

The CONFIG_PREEMPT=n section of setup_singlestep() contains:

    preempt_enable_no_resched();

That's bogus as it is asymetric - no preempt_disable() - and it just
never blew up because preempt_enable_no_resched() is a NOP when
CONFIG_PREEMPT=n. Remove it.

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---
 arch/x86/kernel/kprobes.c |    1 -
 1 file changed, 1 deletion(-)

Index: linux-3.2/arch/x86/kernel/kprobes.c
===================================================================
--- linux-3.2.orig/arch/x86/kernel/kprobes.c
+++ linux-3.2/arch/x86/kernel/kprobes.c
@@ -478,7 +478,6 @@ static void __kprobes setup_singlestep(s
 		 * stepping.
 		 */
 		regs->ip = (unsigned long)p->ainsn.insn;
-		preempt_enable_no_resched();
 		return;
 	}
 #endif
